File: tests/spring/catalog/src/main/java/com/example/catalog/mapper/AuditMapper.java
package com.example.catalog.mapper;

import com.example.catalog.domain.AuditEntry;
import org.apache.ibatis.annotations.Insert;
import org.apache.ibatis.annotations.Mapper;
import org.apache.ibatis.annotations.Select;

@Mapper
public interface AuditMapper {

    @Insert("INSERT INTO audit (action, item_id, created_at) VALUES (#{action}, #{itemId}, #{createdAt})")
    void insert(AuditEntry entry);

    @Select("SELECT COUNT(*) FROM audit")
    long count();
}

File: tests/spring/catalog/expected/src/mapper/audit_mapper.rs
//! Translated from `AuditMapper` (AuditMapper.java) by Java2RustTranslator.

use sqlx::PgConnection;

use crate::domain::AuditEntry;

/// `@Insert`
pub async fn insert(conn: &mut PgConnection, entry: &AuditEntry) -> sqlx::Result<()> {
    sqlx::query("INSERT INTO audit (action, item_id, created_at) VALUES ($1, $2, $3)")
        .bind(&entry.action)
        .bind(entry.item_id)
        .bind(entry.created_at)
        .execute(conn)
        .await?;
    Ok(())
}

/// `@Select`
pub async fn count(conn: &mut PgConnection) -> sqlx::Result<i64> {
    sqlx::query_scalar::<_, i64>("SELECT COUNT(*) FROM audit").fetch_one(conn).await
}
